src/App.jsx

```diff
@@ -14,45 +16,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+export const App = () => {
+  const defaultValue = [...goodsFromServer];
+  const [goods, setGoods] = useState(defaultValue);
+  const [sort, setSort] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  function sortGoods(sortValue) {
+    switch (sortValue) {
+      case 'alphabetically':
+        if (isReversed) {
+          setGoods([...goods]
+            .sort((goodA, goodB) => goodB.localeCompare(goodA)));
+        } else {
+          setGoods([...goods]
+            .sort((goodA, goodB) => goodA.localeCompare(goodB)));
+        }
+
+        break;
+      case 'length':
+        if (isReversed) {
+          setGoods([...goods]
+            .sort((goodA, goodB) => goodB.length - goodA.length));
+        } else {
+          setGoods([...goods]
+            .sort((goodA, goodB) => goodA.length - goodB.length));
+        }
+
+        break;
+      default:
+        setGoods(defaultValue);
+    }
+
+    setSort(sortValue);
+  }
+
+  const handleReverse = function reverse() {
+    setGoods([...goods].reverse());
+  };
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sort !== 'alphabetically',
+          })}
+          onClick={() => sortGoods('alphabetically')}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sort !== 'length',
+          })}
+          onClick={() => sortGoods('length')}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': isReversed === false,
+          })}
+          onClick={() => {
+            setIsReversed(!isReversed);
+            handleReverse();
+          }}
```

Comment:
Just change the state
```suggestion
          onClick={() => {
            setIsReversed(!isReversed);
          }}
```